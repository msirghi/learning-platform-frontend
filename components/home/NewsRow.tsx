import React from "react";
import { News } from "../../common/types";
import styles from "../../styles/modules/Home.module.scss";
import Grid from "@material-ui/core/Grid";
import ArrowRightAltIcon from "@material-ui/icons/ArrowRightAlt";
import IconButton from "@material-ui/core/IconButton";
import dateFormat from "dateformat";

type Props = {
  news: News;
};

export const NewsRow: React.FC<Props> = ({ news: { date, description, title } }) => {
  const formattedDescription =
    description.length > 80 ? `${description.substring(0, 100)}...` : description;

  return (
    <Grid container alignItems='center' spacing={1}  className={styles.cardArrowButton}>
      <Grid item xs={10}>
        <div className={styles.cardNewsTitle}>{title}</div>
        <div className={styles.cardNewsDescription}>{formattedDescription}</div>
        <div>{dateFormat(date, "mediumDate")}</div>
      </Grid>
      <Grid item xs={1}>
        <IconButton>
          <ArrowRightAltIcon color='primary' />
        </IconButton>
      </Grid>
    </Grid>
  );
};
