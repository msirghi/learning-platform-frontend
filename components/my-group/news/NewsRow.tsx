import React from 'react';
import styles from '../../../styles/modules/MyGroup.module.scss';
import dateFormat from 'dateformat';

type Props = {
  title: string;
  description: string;
  date: Date;
  author: string;
};

export const NewsRow: React.FC<Props> = ({ title, description, date, author }) => {
  return (
    <div className={styles.newsRow}>
      <div className={styles.message}>{title}</div>
      <div className={styles.message}>{description}</div>
      <div className={styles.info}>
        <div>{author}</div>
        <div>{dateFormat(date, 'mediumDate')}</div>
      </div>
    </div>
  );
};
