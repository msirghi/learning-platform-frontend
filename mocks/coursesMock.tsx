import { Course } from '../common/types';

export const coursesMock: Course[] = [
  {
    id: '1',
    name: 'Course 1',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Libero non illum est fuga beatae ab, assumenda reiciendis nesciunt numquam similique!`,
    coverImg:
      'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/1200px-Image_created_with_a_mobile_phone.png',
    teacherImg:
      'https://dyl80ryjxr1ke.cloudfront.net/external_assets/hero_examples/hair_beach_v1785392215/original.jpeg'
  },
  {
    id: '2',
    name: 'Course 2',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg: 'https://www.w3schools.com/w3css/img_lights.jpg',
    teacherImg:
      'https://images.pexels.com/photos/614810/pexels-photo-614810.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
  },
  {
    id: '3',
    name: 'Course 3',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg:
      'https://helpx.adobe.com/content/dam/help/en/stock/how-to/visual-reverse-image-search/jcr_content/main-pars/image/visual-reverse-image-search-v2_intro.jpg',
    teacherImg:
      'https://i2-prod.mirror.co.uk/incoming/article14334083.ece/ALTERNATES/s1200c/3_Beautiful-girl-with-a-gentle-smile.jpg'
  },
  {
    id: '4',
    name: 'Course 4',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    coverImg:
      'https://www.metoffice.gov.uk/binaries/content/gallery/metofficegovuk/hero-images/advice/maps-satellite-images/satellite-image-of-globe.jpg',
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '5',
    name: 'Course 5',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '6',
    name: 'Course 6',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '7',
    name: 'Course 7',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '8',
    name: 'Course 8',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '9',
    name: 'Course 9',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '10',
    name: 'Course 10',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '11',
    name: 'Course 11',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '12',
    name: 'Course 12',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  },
  {
    id: '13',
    name: 'Course 13',
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit.`,
    teacherImg: 'https://upload.wikimedia.org/wikipedia/commons/e/ec/Woman_7.jpg'
  }
];
