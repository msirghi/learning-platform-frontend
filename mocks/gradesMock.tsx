import { GradeType } from '../common/enums';
import { Grade } from '../common/types';

export const GradesMock: Grade[] = [
  {
    id: '1',
    course: 'Mathematics',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 1',
    year: 'First year',
    maxPresence: 15,
    presence: 3
  },
  {
    id: '2',
    course: 'Computer Science Computer Science',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.INDIVIDUAL_WORK,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 2',
    year: 'First year',
    maxPresence: 15,
    presence: 7
  },
  {
    id: '3',
    course: 'Biology',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 3',
    year: 'First year',
    maxPresence: 20,
    presence: 15
  },
  {
    id: '4',
    course: 'Physics',
    marks: [
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_1,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.ATTESTATION_2,
        updatedAt: new Date()
      },
      {
        mark: '9.12',
        test: GradeType.EXAM,
        updatedAt: new Date()
      }
    ],
    teacher: 'Teacher 4',
    year: 'First year',
    maxPresence: 20,
    presence: 19
  }
];
