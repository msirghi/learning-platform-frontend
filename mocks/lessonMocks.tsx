import { LessonStatus } from '../common/enums';
import { Lesson } from '../common/types';

export const lessonMock: Lesson[] = [
  {
    id: '22',
    status: LessonStatus.ENDED,
    topic: 'Topic 1',
    title: `Lorem ipsum dolor sit amet.`,
    time: '8:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      10,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      10,
      30,
      0,
      0
    )
  },
  {
    id: '2',
    status: LessonStatus.ONGOING,
    topic: `Lorem ipsum dolor sit amet.`,
    title: 'Lesson 2',
    time: '9:30',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      13,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      14,
      0,
      0,
      0
    )
  },
  {
    id: '3',
    status: LessonStatus.NOT_STARTED,
    topic: 'Topic 3',
    title: 'Lesson 3',
    time: '11:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      14,
      30,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      15,
      30,
      0,
      0
    )
  },
  {
    id: '4',
    status: LessonStatus.NOT_STARTED,
    topic: 'Topic 4',
    title: 'Lesson 4',
    time: '13:00',
    start: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      16,
      0,
      0,
      0
    ),
    end: new Date(
      new Date().getFullYear(),
      new Date().getMonth(),
      new Date().getDate(),
      17,
      30,
      0,
      0
    )
  }
];
