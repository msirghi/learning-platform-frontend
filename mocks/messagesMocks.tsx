import { ChatMessage } from '../common/types';

export const messagesMock: ChatMessage[] = [
  {
    id: 1,
    author: 'Mihail',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet.`
  },
  {
    id: 2,
    author: 'Nicolae',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium, sapiente!`
  },
  {
    id: 3,
    author: 'Mihail 2',
    date: new Date(),
    message: `Lorem ipsum, dolor sit amet consectetur adipisicing.`
  },
  {
    id: 4,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  },
  {
    id: 5,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  },
  {
    id: 6,
    author: 'Mihail 4',
    date: new Date(),
    message: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!`
  }
];
