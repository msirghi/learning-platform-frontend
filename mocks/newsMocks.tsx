import { News } from "../common/types";

export const newsMock: News[] = [
  {
    id: "1",
    date: new Date(),
    description: `Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in.`,
    title: "News 1",
  },
  {
    id: "2",
    date: new Date(),
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae.`,
    title: "News 2",
  },
];
