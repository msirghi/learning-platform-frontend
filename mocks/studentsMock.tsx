import { Student } from '../common/types';

export const studentsMock: Student[] = [
  {
    id: 1,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 2,
    firstName: 'Nicolae',
    lastName: 'Budulae'
  },
  {
    id: 3,
    firstName: 'Budulae',
    lastName: 'Nicolae'
  },
  {
    id: 4,
    firstName: 'Aron',
    lastName: 'Pumnul'
  },
  {
    id: 5,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 6,
    firstName: 'First',
    lastName: 'Last'
  },
  {
    id: 7,
    firstName: 'First',
    lastName: 'Last'
  }
];
