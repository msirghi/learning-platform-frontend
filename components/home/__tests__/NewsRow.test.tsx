import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { newsMock } from '../../../mocks/newsMocks';
import { NewsRow } from '../NewsRow';

describe('NewsRow component', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<NewsRow news={newsMock[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match the snapshot if description length is short', () => {
    const wrapper = shallow(
      <NewsRow
        news={{
          ...newsMock[0],
          description: `Lorem ipsum, dolor.`
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
