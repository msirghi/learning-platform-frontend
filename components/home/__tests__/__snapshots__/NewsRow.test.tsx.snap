// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsRow component should match the snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="cardArrowButton"
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <div
      className="cardNewsTitle"
    >
      News 1
    </div>
    <div
      className="cardNewsDescription"
    >
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debiti...
    </div>
    <div>
      Nov 22, 2000
    </div>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={1}
  >
    <WithStyles(ForwardRef(IconButton))>
      <Memo([object Object])
        color="primary"
      />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`NewsRow component should match the snapshot if description length is short 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="cardArrowButton"
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <div
      className="cardNewsTitle"
    >
      News 1
    </div>
    <div
      className="cardNewsDescription"
    >
      Lorem ipsum, dolor.
    </div>
    <div>
      Nov 22, 2000
    </div>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={1}
  >
    <WithStyles(ForwardRef(IconButton))>
      <Memo([object Object])
        color="primary"
      />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
